### PR TITLE
Adds a "A fin-troduction to `fish`" guide

### PR DESCRIPTION
**Pull request by steven-mathew**
Adds a "A fin-troduction to `fish`" guide.
- Introduction
  - enter `fish`!
  - A dish served best straight out of the oven
  - Installation
- Syntax highlighting
- Autosuggestions
- Autocompletions
- Tab completion
- A `Control`+`R` for the future
- Default shell
- References

### CODE REVIEW

File: Topics/Software_Tools/fish.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
+# A fin-troduction to `fish`
+
```
Comment (allenchen0705) [inline Topics/Software_Tools/fish.md:2]: A table of contents would be helpful!

File: Topics/Software_Tools/fish.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+Some of you might be familiar with zsh plugin managers such as
+[oh-my-zsh](https://ohmyz.sh/) that can configure zsh to your liking, but fish
+just works. No more messing with chunky zsh configs and slow shells that make
+you wait several hundred milliseconds to type another command.
+
+### Installation
+
+It's recommended to check the [fish homepage](https://fishshell.com/) for up-to-date
+instructions for installing the latest version of fish on your platform.
+
```
Comment (allenchen0705) [inline Topics/Software_Tools/fish.md:25]: I installed fish using brew but noticed that your prompt looks different from the default fish prompt I have. How did you get your prompt to look like that?

File: Topics/Software_Tools/fish.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+### enter `fish`!
+
+...the **f**riendly **i**nteractive **sh**ell.
+
+fish offers a command-line interface that is focused on usability and interactive use[^1].
+
+This introduction will be less about the syntax differences between the fish language
+and bash or zsh and more about the out-of-the-box features that are offered from this shell.
+
```
Comment (allenchen0705) [inline Topics/Software_Tools/fish.md:13]: You should briefly define what a shell is.

File: Topics/Software_Tools/fish.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+appear in a different color.
+
+## Autosuggestions
+
+fish suggests commands as you type, based on command history. As you type commands,
+suggestions are offered after the cursor in gray. To accept the suggestion, press
+`→` or `Control`+`F`.
+
+<img width="280" alt="autosuggestions" src="https://github.com/learning-software-engineering/learning-software-engineering.github.io/assets/40929644/fd505f17-4265-41ac-b41f-59bbdd0eb670">
+
```
Comment (allenchen0705) [inline Topics/Software_Tools/fish.md:39]: You could present an example here about the example autocompletions that come with fish.

File: Topics/Software_Tools/fish.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+## A `Control`+`R` for the future
+
+This is a command that you might find yourself using a lot! In bash or zsh, to
+search your history, you can use `Control`+`R` to find some commands you've used
+previously. With fish, you can type a substring, such as 'alac', and press
+`↑` and `↓` (or `Control`+`P` and `Control`+`N`) to traverse commands in your
+history containing 'alac'. You can also use ctrl+r as usual and this opens a pager
+to search your history.
+
```
Comment (allenchen0705) [inline Topics/Software_Tools/fish.md:60]: Just a small nitpick: you use `Control`+`R` but here you are using “ctrl+r”.

File: Topics/Software_Tools/fish.md [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+```bash
+$ echo /usr/local/bin/fish | sudo tee -a /etc/shells
+```
+
+2. Change your default shell:
+
+```bash
+$ chsh -s /usr/local/bin/fish
+```
+
```
Comment (allenchen0705) [inline Topics/Software_Tools/fish.md:77]: I’ve wanted to use fish before but it’s not POSIX compliant and that pushed me away. Add a note about this as a warning to readers.

**Review by allenchen0705 — CHANGES_REQUESTED** [review]
Great work Steven. I left a few comments on changes that I thought would be beneficial to your wiki.

File: Topics/Software_Tools/fish.md [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+search your history, you can use `Control`+`R` to find some commands you've used
+previously. With fish, you can type a substring, such as 'alac', and press
+`↑` and `↓` (or `Control`+`P` and `Control`+`N`) to traverse commands in your
+history containing 'alac'. You can also use ctrl+r as usual and this opens a pager
+to search your history.
+
+## Default shell
+
+At this point, if you are convinced with the benefits of fish, you can change
+your login shell to it as follows:
```
Comment (max-11-max) [inline Topics/Software_Tools/fish.md:64]: Nit: the tone is overall casual and you use “you are” here, but you used “you’re” previously. I suggest changing this to “you’re” to match the tone of the article.

**Review by max-11-max — CHANGES_REQUESTED** [review]
This guide looks great! I left a few comments that might help out a bit.

File: Topics/Software_Tools/fish.md [hunk 8]
```diff
@@ -0,0 +1,80 @@
+# A fin-troduction to `fish`
```
[the remark this answers is not in the document]
Comment (allenchen0705) [reply Topics/Software_Tools/fish.md:1]: Also note that the change needs to be made here in [index.md](https://github.com/learning-software-engineering/learning-software-engineering.github.io/blob/main/index.md)

File: Topics/Software_Tools/fish.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (steven-mathew) [reply Topics/Software_Tools/fish.md:2]: Sounds good, I'll make this change once the sections are all settled.

File: Topics/Software_Tools/fish.md (hunk 2 above)
Comment (steven-mathew) [reply Topics/Software_Tools/fish.md:25]: Yes! I’m using [Tide](https://github.com/IlanCosman/tide). I can add a section about plugins. That’s the only plugin I use with fish.

File: Topics/Software_Tools/fish.md (hunk 5 above)
Comment (steven-mathew) [reply Topics/Software_Tools/fish.md:60]: Ah, thanks for catching this!

File: Topics/Software_Tools/fish.md (hunk 4 above)
Comment (steven-mathew) [reply Topics/Software_Tools/fish.md:39]: Sure, I can include a screenshot in "Autocompletions" as well.

File: Topics/Software_Tools/fish.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (steven-mathew) [reply Topics/Software_Tools/fish.md:1]: Thanks, I missed this.

File: Topics/Software_Tools/fish.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (steven-mathew) [reply Topics/Software_Tools/fish.md:13]: Will do. I'll also add other resources I find here such as modifying colorschemes.

File: Topics/Software_Tools/fish.md [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+### Installation
+
+It's recommended to check the [fish homepage](https://fishshell.com/) for up-to-date
+instructions for installing the latest version of fish on your platform.
+
+## Syntax highlighting
+
+fish performs syntax highlighting as you type. Invalid or non-existent, unexecutable
+commands appear red if they cannot be executed. On the contrary, valid commands will
+appear in a different color.
```
[the remark this answers is not in the document]
Comment (steven-mathew) [reply Topics/Software_Tools/fish.md:30]: Got it, I'll add some screenshots.

File: Topics/Software_Tools/fish.md [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+## Autocompletions
+
+fish comes built with many autocompletions for commands you will encounter often.
+It also parses man pages and automatically generates completions.
+
+## Tab completion
+
+When you type `Tab`, fish guesses the rest of the word under the cursor. It suggests
+a number of possibilities and opens a menu (or "pager") that allows you to select
+the command you're looking for.
```
[the remark this answers is not in the document]
Comment (steven-mathew) [reply Topics/Software_Tools/fish.md:50]: Hmm, I see. I'll add a short sentence at the start of the paragraph for this.

File: Topics/Software_Tools/fish.md (hunk 7 above)
Comment (steven-mathew) [reply Topics/Software_Tools/fish.md:64]: Ah okay, thanks. Will go over the other sections as well to see if there's mismatch in conjunctions to keep the tone consistent.

File: Topics/Software_Tools/fish.md (hunk 3 above)
Comment (steven-mathew) [reply Topics/Software_Tools/fish.md:13]: Thanks for the suggestion. I'll add a brief definition.

File: Topics/Software_Tools/fish.md (hunk 6 above)
Comment (steven-mathew) [reply Topics/Software_Tools/fish.md:77]: That’s right, it isn’t POSIX compliant, but it’s not designed to be. I find that if I need bash for some reason I can always just type bash into the commandline or execute a bash script when I need it. A minor inconvenience for all the benefits that come from using fish. I’ll make a note that mentions this.

**Review by allenchen0705 — COMMENTED** [review]
Great work Steven! Looks good overall. I left comments on some minor changes that you could use.

File: Topics/Software_Tools/fish.md [hunk 11]
```diff
@@ -0,0 +1,145 @@
...
+
+```bash
+$ echo /usr/local/bin/fish | sudo tee -a /etc/shells
+```
+
+2. Change your default shell:
+
+```bash
+$ chsh -s /usr/local/bin/fish
+```
```
[the remark this answers is not in the document]
Comment (steven-mathew) [reply Topics/Software_Tools/fish.md:125]: Thanks, I'll remove them altogehter.

File: Topics/Software_Tools/fish.md [hunk 12]
```diff
@@ -0,0 +1,145 @@
...
+## Syntax highlighting
+
+fish performs syntax highlighting as you type. Invalid or non-existent, unexecutable
+commands appear red if they cannot be executed.
+
+<img width="101" alt="gus" src="https://github.com/learning-software-engineering/learning-software-engineering.github.io/assets/40929644/69c96e62-5404-4f37-840e-1bbdffcb6538">
+
+On the contrary, valid commands will appear in a different color.
+
+<img width="101" alt="git" src="https://github.com/learning-software-engineering/learning-software-engineering.github.io/assets/40929644/46110031-e0e2-4522-b630-791f9c387005">
```
[the remark this answers is not in the document]
Comment (steven-mathew) [reply Topics/Software_Tools/fish.md]: Done!

**Review by max-11-max — APPROVED** [review]
Looks good, this contribution will be helpful!

**Review by allenchen0705 — APPROVED** [review]
Looks good to me! Good work Steven!

**Pull request merged**